include/storage/io.hpp

```diff
@@ -61,6 +62,88 @@ void readHSGR(boost::filesystem::ifstream &input_stream,
 }
 
+// Returns the size of the timestamp in a file
+inline std::uint32_t readTimestampSize(boost::filesystem::ifstream &timestamp_input_stream)
+{
+    timestamp_input_stream.seekg(0, timestamp_input_stream.end);
+    auto length = timestamp_input_stream.tellg();
+    timestamp_input_stream.seekg(0, timestamp_input_stream.beg);
+    return length;
+}
+
+// Reads the timestamp in a file
+inline void readTimestamp(boost::filesystem::ifstream &timestamp_input_stream,
+                          char *timestamp,
+                          std::size_t timestamp_length)
+{
+    timestamp_input_stream.read(timestamp, timestamp_length * sizeof(char));
+}
+
+// Returns the number of edges in a .edges file
+inline std::uint32_t readEdgesSize(boost::filesystem::ifstream &edges_input_stream)
+{
+    std::uint32_t number_of_edges;
+    edges_input_stream.read((char *)&number_of_edges, sizeof(std::uint32_t));
+    return number_of_edges;
+}
+
+// Reads edge data from .edge files which includes its
+// geometry, name ID, turn instruction, lane data ID, travel mode, entry class ID
+template <typename GeometryIDT,
+          typename NameIDT,
+          typename TurnInstructionT,
+          typename LaneDataIDT,
+          typename TravelModeT,
+          typename EntryClassIDT>
+void readEdgesData(boost::filesystem::ifstream &edges_input_stream,
+                   GeometryIDT *geometry_list,
+                   NameIDT *name_id_list,
+                   TurnInstructionT *turn_instruction_list,
+                   LaneDataIDT *lane_data_id_list,
+                   TravelModeT *travel_mode_list,
+                   EntryClassIDT *entry_class_id_list,
+                   std::uint32_t number_of_edges)
+{
+    extractor::OriginalEdgeData current_edge_data;
+    for (std::uint32_t i = 0; i < number_of_edges; ++i)
+    {
+        edges_input_stream.read((char *)&(current_edge_data), sizeof(extractor::OriginalEdgeData));
+        geometry_list[i] = current_edge_data.via_geometry;
+        name_id_list[i] = current_edge_data.name_id;
+        turn_instruction_list[i] = current_edge_data.turn_instruction;
+        lane_data_id_list[i] = current_edge_data.lane_data_id;
+        travel_mode_list[i] = current_edge_data.travel_mode;
+        entry_class_id_list[i] = current_edge_data.entry_classid;
+    }
+}
+
+// Returns the number of nodes in a .nodes file
+inline std::uint32_t readNodesSize(boost::filesystem::ifstream &nodes_input_stream)
+{
+    std::uint32_t number_of_coordinates;
+    nodes_input_stream.read((char *)&number_of_coordinates, sizeof(std::uint32_t));
+    return number_of_coordinates;
+}
+
+// Reads coordinates and OSM node IDs from .nodes files
+// TODO: OSMNodeIDVectorT kind of breaks the consistency of the input parameters
+// Any ideas how to fix this?
+template <typename CoordinateT, typename OSMNodeIDVectorT>
+void readNodesData(boost::filesystem::ifstream &nodes_input_stream,
+                   CoordinateT *coordinate_list,
```

Comment:
`storage.cpp` is using this function to read the nodes data and store it in shared memory e.g. with
```
util::Coordinate *coordinates_ptr = shared_layout_ptr->GetBlockPtr<util::Coordinate, true>(shared_memory_ptr, SharedDataLayout::COORDINATE_LIST);
```
so I am not sure how to do this without plain pointers? Are there ways?